libqpdf/QPDFWriter.cc

```diff
@@ -2451,6 +2476,26 @@ QPDFWriter::writeHeader()
 }
 
 void
+QPDFWriter::writePCLmHeader()
+{
+    setMinimumPDFVersion(pdf.getPDFVersion(), pdf.getExtensionLevel());
+    this->final_pdf_version = this->min_pdf_version;
+    this->final_extension_level = this->min_extension_level;
+    if (! this->forced_pdf_version.empty())
+    {
+	QTC::TC("qpdf", "QPDFWriter using forced PDF version");
+	this->final_pdf_version = this->forced_pdf_version;
+        this->final_extension_level = this->forced_extension_level;
+    }
+
+    writeString("%PDF-");
+    writeString(this->final_pdf_version);
+    // PCLm version
+    writeString("\n%PCLm-1.0\n");
+    writeStringQDF("%QDF-1.0\n\n");
```

Comment:
Let's not forget to make sure fix-qdf works with this or document that it doesn't. My quick reading of fix-qdf suggests that it may require a tweak to make it work.